providers/authn.go

```diff
@@ -0,0 +1,107 @@
+package providers
+
+import (
+	"fmt"
+	"github.com/Flipkart/oauth2_proxy/api"
+	"log"
+	"net/http"
+	"net/url"
+)
+
+type AuthnProvider struct {
+	*ProviderData
+	AuthzUrl *url.URL
+}
+
+func NewAuthnProvider(p *ProviderData) *AuthnProvider {
+	p.ProviderName = "Authn"
+	if p.LoginURL == nil || p.LoginURL.String() == "" {
+		p.LoginURL = &url.URL{
+			Scheme: "http",
+			Host:   "localhost:45101",
+			Path:   "/oauth/authorize",
+		}
+	}
+	if p.RedeemURL == nil || p.RedeemURL.String() == "" {
+		p.RedeemURL = &url.URL{
+			Scheme: "http",
+			Host:   "localhost:45101",
+			Path:   "/oauth/token",
+		}
+	}
+	if p.ProfileURL == nil || p.ProfileURL.String() == "" {
+		p.ProfileURL = &url.URL{
+			Scheme: "http",
+			Host:   "localhost:45101",
+			Path:   "/oauth/r/api/v1/user/details",
+		}
+	}
+	if p.Scope == "" {
+		p.Scope = "user.profile"
+	}
+	return &AuthnProvider{ProviderData: p}
+}
+
+func (p *AuthnProvider) getDetails(s *SessionState) {
+	req, err := http.NewRequest("GET", p.ProfileURL.String(), nil)
+	if err != nil {
+		log.Printf("failed building request %s", err)
```

Comment:
use log "github.com/sirupsen/logrus" for levelled logs. 
and set loglevel using one of the following:
log.SetLevel(log.DebugLevel)
log.SetLevel(log.InfoLevel)
log.SetLevel(log.WarnLevel)